OpenRA.Game/Widgets/ChatDisplayWidget.cs

```diff
@@ -47,7 +47,7 @@ public override void Draw()
 				}
 
 				var text = WidgetUtils.WrapText(line.Text, chatLogArea.Width - inset - 6, font);
-				chatpos.Y -= Math.Max(15, font.Measure(text).Y) + 5;
+				chatpos = chatpos.WithY(pos.Y - Math.Max(15, font.Measure(text).Y) + 5);
```

Comment:
This has to be
```csharp
chatpos = chatpos.WithY(chatpos.Y - Math.Max(15, font.Measure(text).Y) + 5);
```

You can also test it in-game. All the in-game messages outside the chat box are hidden/gone.